From 6b25f9a7be0b5510f055e99e89772816a40876de Mon Sep 17 00:00:00 2001
From: Andreas Beckmann <anbe@debian.org>
Date: Thu, 29 Jan 2026 14:36:25 +0100
Subject: [PATCH] backport conftest.sh comment changes from 515.48.07

---
 conftest.sh | 22 +++++++++++-----------
 1 file changed, 11 insertions(+), 11 deletions(-)

diff --git a/conftest.sh b/conftest.sh
index eabd0f87..6beb5575 100755
--- a/conftest.sh
+++ b/conftest.sh
@@ -545,10 +545,10 @@ compile_test() {
         set_pages_array_uc)
             #
             # Determine if the set_pages_array_uc() function is present.
-            # It does not exist on all architectures. 
-            # 
-            # set_pages_array_uc() was added by commit 
-            # 0f3507555f6fa4acbc85a646d6e8766230db38fc ("x86, CPA: Add 
+            # It does not exist on all architectures.
+            #
+            # set_pages_array_uc() was added by commit
+            # 0f3507555f6fa4acbc85a646d6e8766230db38fc ("x86, CPA: Add
             # set_pages_arrayuc and set_pages_array_wb") in v2.6.30-rc1 (Thu Mar
             # 19 14:51:15 2009)
             #
@@ -1082,7 +1082,7 @@ compile_test() {
             #
             # Determine if mdev_get_type_group_id() function is present or not
             #
-            # Added by commit 15fcc44be0c7a ("vfio/mdev: Add 
+            # Added by commit 15fcc44be0c7a ("vfio/mdev: Add
             # mdev/mtype_get_type_group_id()") in v5.13
             #
             CODE="
@@ -1309,7 +1309,7 @@ compile_test() {
             #
             # The commit c28198889c15 removed the function
             # 'PDE_DATA()', and replaced it with 'pde_data()'
-            # ("proc: remove PDE_DATA() completely") in v5.17-rc1. 
+            # ("proc: remove PDE_DATA() completely") in v5.17-rc1.
             #
             CODE="
             #include <linux/proc_fs.h>
@@ -4024,7 +4024,7 @@ compile_test() {
             #
             # Determine if the 'struct proc_ops' type is present.
             #
-            # Added by commit d56c0d45f0e2 ("proc: decouple proc from VFS with 
+            # Added by commit d56c0d45f0e2 ("proc: decouple proc from VFS with
             # "struct proc_ops"") in 5.6-rc1
             #
             CODE="
@@ -4773,7 +4773,7 @@ compile_test() {
             #
             # Note: KERNELRELEASE and ARCH are defined by Kbuild and automatically
             # passed down to conftest.sh as env vars.
-            
+
             MLNX_OFED_KERNEL_DIR=/usr/src/ofa_kernel
             VAR_DKMS_SOURCES_DIR=$(test -d /var/lib/dkms/mlnx-ofed-kernel &&
                                    ls -d /var/lib/dkms/mlnx-ofed-kernel/*/build 2>/dev/null)
@@ -5171,7 +5171,7 @@ case "$5" in
         #  version gcc 3.2.3
         #
         #  As of this writing, GCC uses a version number as x.y.z and below
-        #  are the typical version strings seen with various distributions. 
+        #  are the typical version strings seen with various distributions.
         #  gcc (GCC) 4.4.7 20120313 (Red Hat 4.4.7-23)
         #  gcc version 4.8.5 20150623 (Red Hat 4.8.5-39) (GCC)
         #  gcc (GCC) 8.3.1 20190507 (Red Hat 8.3.1-4)
@@ -5183,7 +5183,7 @@ case "$5" in
         #
         #  In order to extract GCC version correctly for version strings
         #  like the last one above, we first check for x.y.z and if that
-        #  fails, we fallback to x.y format. 
+        #  fails, we fallback to x.y format.
         VERBOSE=$6
 
         kernel_compile_h=$OUTPUT/include/generated/compile.h
@@ -5480,7 +5480,7 @@ case "$5" in
 
         HASH=$(get_configuration_option CONFIG_MODULE_SIG_HASH)
 
-        if [ $? -eq 0 ] && [ -n $HASH ]; then
+        if [ $? -eq 0 ] && [ -n "$HASH" ]; then
             echo $HASH
             exit 0
         else
-- 
2.39.5

