diff -rup kernel/conftest.sh kernel.fixed/conftest.sh
--- kernel/conftest.sh	2024-05-02 15:25:33.000000000 +0100
+++ kernel.fixed/conftest.sh	2025-10-24 03:16:44.911141863 +0100
@@ -3425,6 +3425,43 @@ compile_test() {
             compile_check_conftest "$CODE" "NV_DMA_BUF_OWNER_PRESENT" "" "types"
         ;;
 
+        drm_fb_create_takes_format_info)
+            #
+            # Determine if a `struct drm_format_info *` is passed into
+            # the .fb_create callback. If so, it will have 4 arguments.
+            # This parameter was added in commit 81112eaac559 ("drm:
+            # Pass the format info to .fb_create") in linux-next
+            # (2025-07-16)
+            CODE="
+            #include <drm/drm_mode_config.h>
+            #include <drm/drm_framebuffer.h>
+
+            static const struct drm_mode_config_funcs funcs;
+            void conftest_drm_fb_create_takes_format_info(void) {
+                funcs.fb_create(NULL, NULL, NULL, NULL); 
+            }"
+
+            compile_check_conftest "$CODE" "NV_DRM_FB_CREATE_TAKES_FORMAT_INFO" "" "types"
+        ;;
+
+        drm_fill_fb_struct_takes_format_info)
+            #
+            # Determine if a `struct drm_format_info *` is passed into
+            # drm_helper_mode_fill_fb_struct(). If so, it will have 4 arguments.
+            # This parameter was added in commit a34cc7bf1034 ("drm:
+            # Allow the caller to pass in the format info to
+            # drm_helper_mode_fill_fb_struct()") in linux-next
+            # (2025-07-16)
+            CODE="
+            #include <drm/drm_modeset_helper.h>
+
+            void conftest_drm_fill_fb_struct_takes_format_info(void) {
+                drm_helper_mode_fill_fb_struct(NULL, NULL, NULL, NULL);
+            }"
+
+            compile_check_conftest "$CODE" "NV_DRM_FILL_FB_STRUCT_TAKES_FORMAT_INFO" "" "types"
+        ;;
+
         drm_connector_funcs_have_mode_in_name)
             #
             # Determine if _mode_ is present in connector function names.  We
Only in kernel.fixed/: conftest.sh.orig
diff -rup kernel/nvidia-drm/nvidia-drm.Kbuild kernel.fixed/nvidia-drm/nvidia-drm.Kbuild
--- kernel/nvidia-drm/nvidia-drm.Kbuild
+++ kernel.fixed/nvidia-drm/nvidia-drm.Kbuild
@@ -126,3 +126,4 @@ NV_CONFTEST_TYPE_COMPILE_TESTS += drm_ou
 NV_CONFTEST_TYPE_COMPILE_TESTS += drm_driver_has_date
 NV_CONFTEST_TYPE_COMPILE_TESTS += file_operations_fop_unsigned_offset_present
 NV_CONFTEST_TYPE_COMPILE_TESTS += drm_connector_helper_funcs_mode_valid_has_const_mode_arg
+NV_CONFTEST_TYPE_COMPILE_TESTS += drm_fb_create_takes_format_info
diff -rup kernel/nvidia-drm/nvidia-drm-drv.c kernel.fixed/nvidia-drm/nvidia-drm-drv.c
--- kernel/nvidia-drm/nvidia-drm-drv.c	2024-05-02 15:50:00.000000000 +0100
+++ kernel.fixed/nvidia-drm/nvidia-drm-drv.c	2025-10-24 03:20:16.568067093 +0100
@@ -135,6 +135,9 @@ static void nv_drm_output_poll_changed(s
 static struct drm_framebuffer *nv_drm_framebuffer_create(
     struct drm_device *dev,
     struct drm_file *file,
+#if defined(NV_DRM_FB_CREATE_TAKES_FORMAT_INFO)
+    const struct drm_format_info *info,
+#endif
     #if defined(NV_DRM_HELPER_MODE_FILL_FB_STRUCT_HAS_CONST_MODE_CMD_ARG)
     const struct drm_mode_fb_cmd2 *cmd
     #else
@@ -150,6 +153,9 @@ static struct drm_framebuffer *nv_drm_fr
     fb = nv_drm_internal_framebuffer_create(
             dev,
             file,
+#if defined(NV_DRM_FB_CREATE_TAKES_FORMAT_INFO)
+            info,
+#endif
             &local_cmd);
 
     #if !defined(NV_DRM_HELPER_MODE_FILL_FB_STRUCT_HAS_CONST_MODE_CMD_ARG)
diff -rup kernel/nvidia-drm/nvidia-drm-fb.c kernel.fixed/nvidia-drm/nvidia-drm-fb.c
--- kernel/nvidia-drm/nvidia-drm-fb.c	2024-05-02 15:50:00.000000000 +0100
+++ kernel.fixed/nvidia-drm/nvidia-drm-fb.c	2025-10-24 03:18:13.527655042 +0100
@@ -184,6 +184,9 @@ static int nv_drm_framebuffer_init(struc
 struct drm_framebuffer *nv_drm_internal_framebuffer_create(
     struct drm_device *dev,
     struct drm_file *file,
+#if defined(NV_DRM_FB_CREATE_TAKES_FORMAT_INFO)
+    const struct drm_format_info *info,
+#endif
     struct drm_mode_fb_cmd2 *cmd)
 {
     struct nv_drm_device *nv_dev = to_nv_device(dev);
@@ -237,6 +240,9 @@ struct drm_framebuffer *nv_drm_internal_
         dev,
         #endif
         &nv_fb->base,
+#if defined(NV_DRM_FB_CREATE_TAKES_FORMAT_INFO)
+        info,
+#endif
         cmd);
 
     /*
Only in kernel.fixed/nvidia-drm: nvidia-drm-fb.c.orig
Only in kernel.fixed/nvidia-drm: nvidia-drm-fb.c.rej
diff -rup kernel/nvidia-drm/nvidia-drm-fb.h kernel.fixed/nvidia-drm/nvidia-drm-fb.h
--- kernel/nvidia-drm/nvidia-drm-fb.h	2024-05-02 15:49:58.000000000 +0100
+++ kernel.fixed/nvidia-drm/nvidia-drm-fb.h	2025-10-24 03:21:14.534261206 +0100
@@ -59,6 +59,9 @@ static inline struct nv_drm_framebuffer
 struct drm_framebuffer *nv_drm_internal_framebuffer_create(
     struct drm_device *dev,
     struct drm_file *file,
+#if defined(NV_DRM_FB_CREATE_TAKES_FORMAT_INFO)
+    const struct drm_format_info *info,
+#endif
     struct drm_mode_fb_cmd2 *cmd);
 
 #endif /* NV_DRM_ATOMIC_MODESET_AVAILABLE */
Only in kernel.fixed/nvidia-drm: nvidia-drm-fb.h.orig
Only in kernel.fixed/nvidia-drm: nvidia-drm-fb.h.rej
